nncf/experimental/tensorflow/patch_tf.py

```diff
@@ -0,0 +1,290 @@
+        self._ops_weights = ops_weights
+
+        arg_provider_cls = TF_ARG_PROVIDERS.registry_dict.get(self._target_point.op_type_name)
+        if arg_provider_cls is None:
+            raise ValueError(f'Unexpected type of the TensorFlow operation: {self._target_point.op_type_name}. '
+                             'Register an `ArgProvider` instance for this type in the '
+                             '`TF_ARG_PROVIDERS` registry, please.')
+
+        self._arg_provider = arg_provider_cls()
+
+    @property
+    def operations(self) -> List[NNCFOperation]:
+        return self._operations
+
+    @property
+    def target_point(self) -> TFTargetPoint:
+        return self._target_point
+
+    @property
+    def is_pre_hook(self) -> bool:
+        """
+        Returns the boolean flag that specified whether the hook should be
+        applied before or after target operation.
+
+        :return: `True` if hook should be applied before target operation,
+            `False` otherwise.
+        """
+        return self._target_point.type == TargetType.OPERATOR_PRE_HOOK
+
+    def get_operation_weights(self, op_name: str) -> Any:
+        """
+        Returns weights of the operation with `op_name`.
+
+        :param op_name: Name of the operation.
+        :return: Weihts of the operation.
+        """
+        return self._ops_weights[op_name]
+
+    def __call__(self, *args, **kwargs):
+        """
+        Applies this hook.
+
+        :return: A tuple (args, kwargs)
+        """
+        if self.is_pre_hook:
+            get_fn = self._arg_provider.get_input
+            set_fn = self._arg_provider.set_input
+        else:
+            get_fn = self._arg_provider.get_output
+            set_fn = self._arg_provider.set_output
+
+        x = get_fn(self.target_point.port_id, args, kwargs)
+        for op in self.operations:
+            w = self.get_operation_weights(op.name)
+            x = op(*(x, w, None))
+        args_, kwargs_ = set_fn(self.target_point.port_id, x, args, kwargs)
+
+        return args_, kwargs_
+
+
+class TensorFlowOpWrapper:
+    """
+    Describes a wrapper around the method from the TensorFlow API.
+    """
+
+    def __init__(self, op, op_type_name: str):
+        """
+        Initializes a wrapper.
+
+        :param op: Original method.
+        :param op_type_name: Operation type name (name of the function
+            from the `tf.raw_ops` module).
+        """
+        self._op = op
+        self._op_type_name = op_type_name
+
+    def __call__(self, *args, **kwargs):
+        """
+        Applies TensorFlow operation with compression extensions.
+        """
+        tracing_context = get_current_context()
+
+        # Should we wrap current operation?
+        if not tracing_context.wrap_ops:
+            return self._op(*args, **kwargs)
+
+        op_name = get_op_name(self._op_type_name, kwargs.get('name'))
+
+        _pre_hooks = getattr(get_current_context().model, '_pre_hooks')
```

Comment:
formally model can be None, try to avoid this by explicitly passing it to `enter` method

Reply:
Thanks for the comment! I fixed it.